code/obj/item/device/lights.dm

```diff
@@ -735,3 +735,158 @@ TYPEINFO(/obj/item/device/light/floodlight)
 	rotatable = FALSE
 	infinite_power = TRUE
 	power_usage = 0 WATTS
+
+#define FLARE_UNLIT 1
+#define FLARE_LIT 2
+#define FLARE_BURNT 3
+
+/obj/item/roadflare
+	name = "emergency flare"
+	desc = "Space grade emergency flare that can burn in an 02 free environment. Estimated burn time 3-6 minutes."
+	icon = 'icons/obj/lighting.dmi'
+	inhand_image_icon = 'icons/mob/inhand/hand_tools.dmi'
+	icon_state = "roadflare"
+	uses_multiple_icon_states = 1
+	w_class = W_CLASS_POCKET_SIZED
+	throwforce = 1
+	flags = FPRINT | TABLEPASS
+	stamina_damage = 0
+	stamina_cost = 0
+	stamina_crit_chance = 1
+	burn_point = 220
+	burn_output = 1200
+	burn_possible = 1
+
+	var/on = FLARE_UNLIT
+
+	var/life_time = 0
+	rand_pos = 1
+
+	var/col_r = 0.95
+	var/col_g = 0.7
+	var/col_b = 0.25
+	var/brightness = 0.4
+	var/datum/component/loctargeting/medium_light/light_c
```

Comment:
Ideally you shouldn't be storing references to components, if you want it to start off then there's a bool argument for that, and it can be turned on and off by sending `COMSIG_LIGHT_ENABLE` and `COMSIG_LIGHT_DISABLE` respectively